libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -234,6 +235,36 @@ void HTTPClient::end(void)
     }
 }
 
+/* support FirebaseStream - Type, Event */
+String HTTPClient::getType(void)
+{
+	String type = _tcp->readStringUntil('\n').substring(7); //put
+	return type;
+}
+
+String HTTPClient::getEvent(void)
+{
+	String event = _tcp->readStringUntil('\n').substring(6); //{"path":"/","data":"416.28"}	
```

Comment:
This is something specific to Firebase, and does not make much sense in the context of plain HTTP protocol. I suggest making a class derived from HTTPClient which would have these Firebase-specific changes.